File: my-folder/0049-group-anagrams/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagramMap = new HashMap<>();
        for (String s : strs) {
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String sortedKey = new String(charArray);

            anagramMap.computeIfAbsent(sortedKey, k -> new ArrayList<>()).add(s);
        }
        return new ArrayList<>(anagramMap.values());
    }
}

File: my-folder/0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        freqs = {}
        for s in strs:
            key = "".join(sorted(s))
            if key not in freqs:
                freqs[key] = []
            freqs[key].append(s)
        
        ans = []
        for key in freqs:
            ans.append(freqs[key])
        
        return ans
